datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -2820,4 +2993,247 @@ mod tests {
+                StringArray::from(vec!["05:08:26", "05::08::27"]),
+            ),
+            (
+                Arc::new(Time64MicrosecondArray::from(vec![12344567000, 22244567000]))
+                    as ArrayRef,
+                StringArray::from(vec!["%H:%M:%S", "%H::%M::%S"]),
+                StringArray::from(vec!["03:25:44", "06::10::44"]),
+            ),
+            (
+                Arc::new(Time64NanosecondArray::from(vec![
+                    1234456789000,
+                    2224456789000,
+                ])) as ArrayRef,
+                StringArray::from(vec!["%H:%M:%S", "%H::%M::%S"]),
+                StringArray::from(vec!["00:20:34", "00::37::04"]),
+            ),
+            (
+                Arc::new(TimestampSecondArray::from(vec![
+                    date.timestamp(),
+                    date2.timestamp(),
+                ])) as ArrayRef,
+                StringArray::from(vec!["%Y::%m::%d %S::%M::%H", "%d::%m::%Y %S-%M-%H"]),
+                StringArray::from(vec![
+                    "2020::01::02 05::04::03",
+                    "08::07::2026 11-10-09",
+                ]),
+            ),
+            (
+                Arc::new(TimestampMillisecondArray::from(vec![
+                    date.timestamp_millis(),
+                    date2.timestamp_millis(),
+                ])) as ArrayRef,
+                StringArray::from(vec![
+                    "%Y::%m::%d %S::%M::%H %f",
+                    "%d::%m::%Y %S-%M-%H %f",
+                ]),
+                StringArray::from(vec![
+                    "2020::01::02 05::04::03 000000000",
+                    "08::07::2026 11-10-09 000000000",
+                ]),
+            ),
+            (
+                Arc::new(TimestampMicrosecondArray::from(vec![
+                    date.timestamp_micros(),
+                    date2.timestamp_micros(),
+                ])) as ArrayRef,
+                StringArray::from(vec![
+                    "%Y::%m::%d %S::%M::%H %f",
+                    "%d::%m::%Y %S-%M-%H %f",
+                ]),
+                StringArray::from(vec![
+                    "2020::01::02 05::04::03 000012000",
+                    "08::07::2026 11-10-09 000056000",
+                ]),
+            ),
+            (
+                Arc::new(TimestampNanosecondArray::from(vec![
+                    date.timestamp_nanos_opt().unwrap(),
+                    date2.timestamp_nanos_opt().unwrap(),
+                ])) as ArrayRef,
+                StringArray::from(vec![
+                    "%Y::%m::%d %S::%M::%H %f",
+                    "%d::%m::%Y %S-%M-%H %f",
+                ]),
+                StringArray::from(vec![
+                    "2020::01::02 05::04::03 000012345",
+                    "08::07::2026 11-10-09 000056789",
+                ]),
+            ),
+        ];
+
+        for (value, format, expected) in array_scalar_data {
+            let result = to_char(&[
```

Comment:
These are good test cases 👍 

I think we should add tests with:
1. The dates as a scalar, and the format as an array
2. Two scalar arguments

Reply:
The test_to_char covers scalar/scalar, array/scalar (single test) and array/array currently. I'll add a test for scalar/array